sources/osgAnimation/UpdateMorph.js

```diff
@@ -47,21 +56,174 @@ define( [
+                    var vAttrs = morph.getVertexAttributeList();
+                    var morphNames = morph.getMorphTargetNames();
+                    for ( var k = 0, nbNames = morphNames.length; k < nbNames; ++k ) {
+
+                        var attName = morphNames[ k ];
+                        vAttrs[ attName + '_' + i ].setActiveArray( vAttrs[ attName + '_' + index ] );
+
+                    }
+                }
+            }
+        },
+
+        _mergeExtraMorphTarget: function ( attrs, attName, extraWeightSum ) {
+
+            var i = 0;
+            // ignore the gpu morphed
+            var gpuMorphed = this._gpuMorphed;
+            var vAttr = attrs[ attName ];
+            var vertexLen = vAttr.getElements().length;
+            var nbVertex = vertexLen / vAttr.getItemSize();
+            var itemSize = vAttr.getItemSize();
+
+            vAttr._cpuMorph = vAttr._cpuMorph || new BufferArray( BufferArray.ARRAY_BUFFER, new Float32Array( vertexLen ), itemSize );
+            var morphExtraTargets = vAttr._cpuMorph.getElements();
+            for ( i = 0; i < vertexLen; ++i ) {
+                morphExtraTargets[ i ] = 0.0;
+            }
+
+            var targets = this._targets;
+            for ( var j = 0, nb = targets.length; j < nb; ++j ) {
+                var weight = targets[ j ].value;
+                if ( gpuMorphed[ j ] === true ) continue; // ignore gpu morphed targets
+                if ( Math.abs( weight ) < WEIGHT_CPU_EPS ) continue;
+
+                weight /= extraWeightSum;
+
+                var morphElts = attrs[ attName + '_' + j ].getElements();
+                for ( i = 0; i < nbVertex; ++i ) {
+
+                    var k = i * itemSize;
+                    morphExtraTargets[ k ] += weight * morphElts[ k ];
+                    morphExtraTargets[ k + 1 ] += weight * morphElts[ k + 1 ];
+                    morphExtraTargets[ k + 2 ] += weight * morphElts[ k + 2 ];
+                    // don't morph tangent w component
+                }
+            }
+
+            // map on last index target
+            attrs[ attName + '_3' ].setActiveArray( vAttr._cpuMorph );
+            vAttr._cpuMorph.dirty();
+        },
+
+        _computeExtraWeightsSum: function () {
+            var gpuMorphed = this._gpuMorphed;
+            var sum = 0.0;
+            var targets = this._targets;
+            for ( var i = 0, nb = targets.length; i < nb; ++i ) {
+                var weight = targets[ i ].value;
+                if ( gpuMorphed[ i ] === true ) continue; // ignore gpu morphed targets
+                if ( Math.abs( weight ) < WEIGHT_CPU_EPS ) continue;
+                sum += weight;
+            }
+            return sum;
+        },
+
+        _morphBufferArrayCPU: function () {
+
+            // compute new weights of the 4th target (all the extra target will be merged inside this one)
+            var extraWeightSum = this._weights[ 3 ] = this._computeExtraWeightsSum();
```

Comment:
it can't be zero, at least we have https://github.com/cedricpinson/osgjs/blob/morph-many-targets/sources/osgAnimation/UpdateMorph.js#L214